src/NuGet.Core/NuGet.Protocol/PackageDownloadContext.cs

```diff
@@ -37,12 +38,23 @@ public class PackageDownloadContext
             DirectDownloadDirectory = directDownloadDirectory;
         }
 
+        public PackageDownloadContext(
+            SourceCacheContext sourceCacheContext,
+            string directDownloadDirectory,
+            bool directDownload,
+            SearchTree searchTree)
```

Comment:
I'd use the configuration object or an equivalent thing that communicates that it contains namespaces information.

Reply:
Ok. Now passing down `ISettings` object instead of just SearchTree.